crates/core/src/task_center/handle.rs

```diff
@@ -57,9 +59,30 @@ impl Handle {
     /// Attempt to set the global metadata handle. This should be called once
     /// at the startup of the node.
     pub fn try_set_global_metadata(&self, metadata: Metadata) -> bool {
+        self.inner
+            .health
+            .node_status()
+            .merge(NodeStatus::StartingUp);
```

Comment:
The `NodeStatus` is spread across the code base with this change. But I guess w/o structured concurrency there is no way to keep it in fewer places.